openedx/core/djangoapps/olx_rest_api/block_serializer.py

```diff
@@ -0,0 +1,159 @@
+        # Add  <xblock-include /> tags for each child (XBlock XML export
+        # normally puts children inline as e.g. <html> tags, but we want
+        # references to them only.)
+        if block.has_children:
+            for child_id in block.children:
+                # In modulestore, the "definition key" is a MongoDB ObjectID
+                # kept in split's definitions table, which theoretically allows
+                # the same block to be used in many places (each with a unique
+                # usage key). However, that functionality is not exposed in
+                # Studio (other than via content libraries). So when we import
+                # into Blockstore, we assume that each usage is unique, don't
+                # generate a usage key, and create a new "definition key" from
+                # the original usage key.
+                # So modulestore usage key
+                #     block-v1:A+B+C+type@html+block@introduction
+                # will become Blockstore definition key
+                #     html+introduction
+                #
+                # If we needed the real definition key, we could get it via
+                #     child = block.runtime.get_block(child_id)
+                #     child_def_id = str(child.scope_ids.def_id)
+                # and then use
+                #     <xblock-include definition={child_def_id} usage={child_id.block_id} />
+                def_id = blockstore_def_key_from_modulestore_usage_key(child_id)
+                olx_node.append(olx_node.makeelement("xblock-include", {"definition": def_id}))
+        # Store the resulting XML as a string:
+        self.olx_str = etree.tostring(olx_node, encoding="unicode", pretty_print=True)
+
+    def serialize_html_block(self, block):
+        """
+        Special case handling for HTML blocks
+        """
+        olx_node = etree.Element("html")
+        if block.display_name:
+            olx_node.attrib["display_name"] = block.display_name
+        olx_node.text = etree.CDATA("\n" + block.data + "\n")
+        self.olx_str = etree.tostring(olx_node, encoding="unicode", pretty_print=True)
+
+    def transform_olx(self, olx_node):
+        """
+        Apply transformations to the given OLX etree Node.
+        """
+        # Remove 'url_name' - we store the definition key in the folder name
+        # that holds the OLX and the usage key elsewhere, so specifying it
+        # within the OLX file is redundant and can lead to issues if the file is
+        # copied and pasted elsewhere in the bundle with a new definition key.
+        olx_node.attrib.pop('url_name', None)
+        # Convert <vertical> to the new <unit> tag/block
+        if olx_node.tag == 'vertical':
+            olx_node.tag = 'unit'
+            for key in olx_node.attrib.keys():
+                if key not in ('display_name', 'url_name'):
+                    log.warning('<vertical> tag attribute "%s" will be ignored after conversion to <unit>', key)
```

Comment:
If you're logging this, please log some kind of identifier so that we can trace back the original offending content.

Reply:
Fixed in cd17b43a0f939d003521f23393154b21adb0edcd